# Re-run of the analysis using an updated transcriptome

Using a transcriptome annotation generated by Dr Steve Doyle, 22nd April 2020 WBPS15

**All bases used were at least MAPQ20**
	

As before, to determine the SNP rate per CDS
	
Want to have the total number of SNPs with AF >40% cf ref genome

And the total length of the CDS.
	
1. To determine the new length of each CDS:
	

To get the total length of the CDS:
	
Use the gff3 file.
	
Check file format:
	
```
	head HCON_V4_curated_20200422_WBPS15.gff3
	##gff-version 3
	##sequence-region       hcontortus_chr1_Celeg_TT_arrow_pilon    1       45774638
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       gene    5834    6930    .       +       .       Name=HCON_00000010;ID=e3acd612-2fe0-4ed8-af87-7af3c1c1ef7e;date_creation=2020-03-31;date_last_modified=2020-03-31
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       pseudogenic_transcript  5834    6930    .       +       .       Name=HCON_00000010-00001;ID=fe4a5e17-36f5-4409-98a1-ccce8f3af1de;Parent=e3acd612-2fe0-4ed8-af87-7af3c1c1ef7e;date_creation=2020-03-31;date_last_modified=2020-03-31
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       exon    17482   18190   .       -       .       Name=23ebe882-f6ba-4c2b-bf6c-e1320db1a308;ID=23ebe882-f6ba-4c2b-bf6c-e1320db1a308;Parent=HCON_00000020-00001
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       gene    17482   28800   .       -       .       Name=HCON_00000020;ID=caada99d-170f-4735-84fa-633532d2de25;biotype=protein_coding;date_creation=2019-02-27;date_last_modified=2019-02-27;previous_stable_id=HCOI01207600
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       mRNA    17482   28800   .       -       .       Name=HCON_00000020-00001;ID=HCON_00000020-00001;Parent=caada99d-170f-4735-84fa-633532d2de25;date_creation=2019-02-27;date_last_modified=2019-02-27
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       CDS     18085   18190   .       -       1       Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=22414322-dd92-48c1-832c-c3a2523c584d;Parent=HCON_00000020-00001
	
```
Ok, the format of the Name/ID/Parent column is DIFFERENT (wildly so!!!!) from before. 
	
So, before transcripts would be **.t1** or **.t2**, and gene names (with no transcripts) would not have any .t
	
Now, transcripts are **Name=HCON_00000290-00001** and Name=HCON_00000290-00002 
	
And, I think that even if there is no second transcript, all are -00001. 
	

First extract the CDS lines only. 
To get just the first transcript (-00001). 
	
Assume (Note that the .gff3 file has the gene name alone if a single transcript, or with a .1 or .2 etc if more than one transcript).
	
To add 4 extra dots to the grep -v to collect only those genes with a single transcript or the first transcript.
	

Note, for ease of use, I am keeping the output file as essentially the same name as before, just changed to WBPS15. But it has not been uploaded to WBPS yet!
	
```
	find . -name "HCON_V4_curated_20200422_WBPS15.gff3" | xargs grep -E 'CDS' | grep -v 'ID=cds:HCON_.............[2-9]' > CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	
```
Get:
```
more CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       CDS     18085   18190   .       -       1       Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=2241432
	2-dd92-48c1-832c-c3a2523c584d;Parent=HCON_00000020-00001
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       CDS     23328   23452   .       -       0       Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=2241432
	2-dd92-48c1-832c-c3a2523c584d;Parent=HCON_00000020-00001
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       CDS     24224   24319   .       -       0       Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=2241432
	2-dd92-48c1-832c-c3a2523c584d;Parent=HCON_00000020-00001
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       CDS     28633   28794   .       -       0       Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=2241432
	2-dd92-48c1-832c-c3a2523c584d;Parent=HCON_00000020-00001
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       CDS     31561   31707   .       +       0       Name=acdfe94b-b140-4b19-9032-c4413a4c055d;ID=acdfe94
	b-b140-4b19-9032-c4413a4c055d;Parent=HCON_00000030-00001
	hcontortus_chr1_Celeg_TT_arrow_pilon    WormBase_imported       CDS     32036   32192   .       +       0       Name=acdfe94b-b140-4b19-9032-c4413a4c055d;ID=acdfe94
	b-b140-4b19-9032-c4413a4c055d;Parent=HCON_00000030-00001
	
```
	

Good (remember that the first gene had no CDS as it was a pseudogene, hence why the above starts with ...20)
	

Therefore, to determine length of each CDS (exons excluding UTRs) do:
	
```
awk '{print $1 "\t" $3 "\t" $4 "\t" $5 "\t" $9 "\t" ($5 - $4) }' CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3 > length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	
```
	
Get the length in basepairs of each CDS as a new column:
```
more length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     18085   18190   Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=22414322-dd92-48c1-832c-c3a2523c584d;Parent=HCON_000000
	20-00001        105
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     23328   23452   Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=22414322-dd92-48c1-832c-c3a2523c584d;Parent=HCON_000000
	20-00001        124
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     24224   24319   Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=22414322-dd92-48c1-832c-c3a2523c584d;Parent=HCON_000000
	20-00001        95
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     28633   28794   Name=22414322-dd92-48c1-832c-c3a2523c584d;ID=22414322-dd92-48c1-832c-c3a2523c584d;Parent=HCON_000000
	20-00001        161
```	

To get the gene number/name:

Need to figure out how to remove all but the gene name..... 

Using the metacharacters ``` .* ``` in that order removes everything until you tell it to stop - it's like ``` . ``` multiple times. 
	
```
sed 's/Name=.*;ID=.*;Parent=//g' length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3  > gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	
```

So now have the length of each exon in nucleotides. And the gene name for each gene associated with each exon. 
	
```
	more gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     18085   18190   HCON_00000020-00001     105
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     23328   23452   HCON_00000020-00001     124
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     24224   24319   HCON_00000020-00001     95
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     28633   28794   HCON_00000020-00001     161
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     31561   31707   HCON_00000030-00001     146
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     32036   32192   HCON_00000030-00001     156
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     32266   32405   HCON_00000030-00001     139
	hcontortus_chr1_Celeg_TT_arrow_pilon    CDS     32465   32568   HCON_00000030-00001     103
```	

	

Now I want to sum the length of each gene.
Note, the following produces an array using the values in column 5 (the gene name) and summing the values in column 6 (the exon lengths). It also counts the number of exons and the output file contains: Gene name "\t" CDS length "\t" Total number of exons. 
	

```
awk -F "\t" '{a[$5] += $6; ++c[$5]} END{for (i in a) print i, a[i], c[i]}' gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3 > SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	
#The -F is the field separator. 
```	

Get:
```
more SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	HCON_00151190-00001 953 7
	HCON_00136280-00001 514 7
	HCON_00130730-00001 1001 10
	HCON_00115820-00001 431 1
	HCON_00009880-00001 1926 21
	HCON_00024790-00001 1164 9
	HCON_00135615-00001 496 2
```	

	
Note, for some reason genes are not in same order as in the .gff3 file, but checked and seem to be ok with the answers!

```
grep 'HCON_00000020-00001' SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3 > check
more check
HCON_00000020-00001 485 4
```	

	

Need to get the CDS file so that can make into a .bed file, in order to merge with the mpileup file to determine SNPs. 
Need to get the first column to be just the chromosome to make into a .bed file
	
```
sed 's/hcontortus_chr//g' gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3 > Hcon_chr_1 


sed 's/_Celeg_TT_arrow_pilon//g' Hcon_chr_1 > Hcon_chr_2
```	

Get:

```
more Hcon_chr_2
	1       CDS     18085   18190   HCON_00000020-00001     105
	1       CDS     23328   23452   HCON_00000020-00001     124
	1       CDS     24224   24319   HCON_00000020-00001     95
	1       CDS     28633   28794   HCON_00000020-00001     161
	1       CDS     31561   31707   HCON_00000030-00001     146
	1       CDS     32036   32192   HCON_00000030-00001     156
```	

	
To get to right format do:
	
```
awk '{print $1 "\t" $3 "\t" $4 "\t" $2 "\t" $5 "\t" $6}' Hcon_chr_2  > gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.bed
```	

Get: 
```
more gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.bed
	1       18085   18190   CDS     HCON_00000020-00001     105
	1       23328   23452   CDS     HCON_00000020-00001     124
	1       24224   24319   CDS     HCON_00000020-00001     95
	1       28633   28794   CDS     HCON_00000020-00001     161
	1       31561   31707   CDS     HCON_00000030-00001     146
	1       32036   32192   CDS     HCON_00000030-00001     156
```	

	

#######################Files generated: ################
Input file (from Steve):
	HCON_V4_curated_20200422_WBPS15.gff3
	Sum of CDS length per gene:
	-rw-rw-r-- 1 jmi45g jmi45g   569600 Jun  4 14:14 SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	Each CDS with length as a bed file for merging with the mpileup file to obtain SNPS:
	-rw-rw-r-- 1 jmi45g jmi45g  8884725 Jun  4 14:23 gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.bed
	

	

	

	###########################################################
	

	2. To obtain from the MAPQ20 mpileup file (created previously (see line 830 in ... ) only those variants within the coding sequence
	The previous MAPQ20 mpileup file (.bed format) is called MAPQ20.UGA_ISE.mpileup2 
	

	To get only those sites that I am interested in: 
	

	Make a bed file of the mpileup file:
	

	To make into a .bed file to extract the CDS:
	

	#!/bin/sh
	#PBS -l walltime=50:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	awk '{print $1 "\t" $2 "\t" ($2+1) "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9}' /export/home/jmi45g/Hcont_Snp_diversity/MAPQ20.UGA_ISE.mpileup2 > /export/home/jmi45g/Hcont_Snp_diversity/bp2_MAPQ20.UGA_ISE.mpileup2
	

	On the HPCC - paste into nano. 
	Save as run file
	chmod +x run_file_name
	RE-OPEN in nano the run file and make sure that all the lines are correct!
	It tends to make the first 4 one line for some reason and chop up the rest. So annoying. 
	Re-save
	

	Run on the HPCC using:
	qsub ./run_file_name
	

	To monitor progress of a job:
	qstat
	

	To make into a bed file:
	

	#!/bin/sh
	#PBS -l walltime=50:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	sed 's/hcontortus_chr//g' /export/home/jmi45g/Hcont_Snp_diversity/bp2_MAPQ20.UGA_ISE.mpileup2 > /export/home/jmi45g/Hcont_Snp_diversity/Hcon_chr_1 
	

	#!/bin/sh
	#PBS -l walltime=50:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	sed 's/_Celeg_TT_arrow_pilon//g' /export/home/jmi45g/Hcont_Snp_diversity/Hcon_chr_1 > /export/home/jmi45g/Hcont_Snp_diversity/bp2_MAPQ20.UGA_ISE.mpileup2.bed
	

	Note, it still has the mtDNA written out a bit more. Not sure if this will be a problem. 
	Don't remember doing anything about it last time!
	

	#!/bin/sh
	#PBS -l walltime=50:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	bedtools intersect -a /export/home/jmi45g/Hcont_Snp_diversity/bp2_MAPQ20.UGA_ISE.mpileup2.bed -b /export/home/jmi45g/Hcont_Snp_diversity/gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.bed -wa -wb -f 2E-9 > /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.mpileup2.bed
	

	All fine - it merged no problem, but note that there is no mt_DNA in it. 
	

	To change this back into an mpileup file for popoolation2:
	Note, am removing the gene file info and ALSO the second bp column!! 
	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	awk '{print $1 "\t" $2 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 "\t" $10}' /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.mpileup2.bed > /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.mpileup
	

	For the following steps use PoPoolation2 (Kofler et al., 2011b). 
	To identify indels of 2 basepairs or greater:
	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	perl  /export/home/jmi45g/popoolation2_1201/indel_filtering/identify-indel-regions.pl \
	--indel-window 5 \
	--min-count 2 \
	--input /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.mpileup \
	--output /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.mpileup.indels.gtf
	

	To sync the filtered mpileup file (discarding low quality bases):
	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	java -ea -Xmx7g -jar /export/home/jmi45g/popoolation2_1201/mpileup2sync.jar \
	--input /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.mpileup \
	--output /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.java.sync \
	--fastq-type sanger \
	--min-qual 20 \
	--threads 8
	

	To then filter the basepairs around the indels (as these can have false SNPs due to poor alignment of reads 
	in these positions (Schlotterer et al., 2014) - an alternative is to re-align the reads, as GATK does):
	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	perl /export/home/jmi45g/popoolation2_1201/indel_filtering/filter-sync-by-gtf.pl \
	--input /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.java.sync \
	--gtf /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.mpileup.indels.gtf \
	--output /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.noindels.java.sync
	

	To then make separate UGA and ISE files:
	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	awk '{print $1 "\t" $2 "\t" $3 "\t" $4}' /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.noindels.java.sync > /export/home/jmi45g/Hcont_Snp_diversity/UGA_CDS_WBPS15_bp2_MAPQ20.UGA_ISE.noindels.java.sync
	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	awk '{print $1 "\t" $2 "\t" $3 "\t" $5}' /export/home/jmi45g/Hcont_Snp_diversity/CDS_WBPS15_bp2_MAPQ20.UGA_ISE.noindels.java.sync > /export/home/jmi45g/Hcont_Snp_diversity/ISE_CDS_WBPS15_bp2_MAPQ20.UGA_ISE.noindels.java.sync
	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	perl /export/home/jmi45g/popoolation2_1201/snp-frequency-diff.pl \
	--input /export/home/jmi45g/Hcont_Snp_diversity/UGA_CDS_WBPS15_bp2_MAPQ20.UGA_ISE.noindels.java.sync \
	--output-prefix /export/home/jmi45g/Hcont_Snp_diversity/UGA_CDS_WBPS15_MAPQ20. \
	--min-count 2 \
	--min-coverage 10 \
	--max-coverage 2%
	

	#Result: '--max-coverage 2%' is equivalent to 
	

	

	#!/bin/sh
	#PBS -l walltime=30:00:00
	#PBS -l cput=50:00:00
	#PBS -l nodes=1:centos6
	perl /export/home/jmi45g/popoolation2_1201/snp-frequency-diff.pl \
	--input /export/home/jmi45g/Hcont_Snp_diversity/ISE_CDS_WBPS15_bp2_MAPQ20.UGA_ISE.noindels.java.sync \
	--output-prefix /export/home/jmi45g/Hcont_Snp_diversity/ISE_CDS_WBPS15_MAPQ20. \
	--min-count 2 \
	--min-coverage 10 \
	--max-coverage 2%
	

	Result: 
	

	To get biallelic sites only: 
	

	awk '$4 == 2' UGA_CDS_WBPS15_MAPQ20._rc > biallelic_UGA_CDS_WBPS15_MAPQ20._rc
	awk '$4 == 2' ISE_CDS_WBPS15_MAPQ20._rc > biallelic_ISE_CDS_WBPS15_MAPQ20._rc
	

	

	########################################################################################################################
	####MAPQ WBPS15 20 UGA ######################################################################################################
	

	Then in R v 3.3.3 (2017-03-06) -- "Another Canoe"
	

	af<-read.table("biallelic_UGA_CDS_WBPS15_MAPQ20._rc", header=F, sep="\t", na.strings="")
	summary(af)
	

	>  V1               V2           V3               V4          V5
	 1:377853   Min.   :    2379   A:467482   Min.   :2   G/A    :349255
	 2:300277   1st Qu.:10387541   C:451939   1st Qu.:2   C/T    :347958
	 3:365262   Median :24633377   G:453036   Median :2   T/C    :343963
	 4:351117   Mean   :24004467   T:467070   Mean   :2   A/G    :343766
	 5:330512   3rd Qu.:37243744              3rd Qu.:2   A/T    : 77055
	 X:114506   Max.   :51822498              Max.   :2   T/A    : 76503
	                                                      (Other):301027
	       V6         V7          V8         V9              V10
	 Min.   :0   pop   :1578641   A:469511   A:485261   81/83  :   4329
	 1st Qu.:0   rc    :     58   C:450250   C:435310   88/90  :   4238
	 Median :0   rc|pop: 260828   G:451357   G:434498   84/86  :   4231
	 Mean   :0                    T:468409   T:484458   85/87  :   4218
	 3rd Qu.:0                                          90/92  :   4217
	 Max.   :0                                          86/88  :   4213
	                                                    (Other):1814081
	      V11
	 2/83   :   4460
	 2/87   :   4356
	 2/90   :   4350
	 2/86   :   4346
	 2/88   :   4341
	 2/94   :   4323
	 (Other):1813351
	

	

	To make the major and minor allele frequencies numerical and not factors...
	

	So, first:
	af$V10<-as.character(af$V10)
	af$V10<-as.numeric( gsub("(\\d+)/(\\d+)", "\\1", af$V10, perl=T) ) / as.numeric ( gsub("(\\d+)/(\\d+)", "\\2", af$V10, perl=T) ) 
	

	af$V11<-as.character(af$V11)
	af$V11<-as.numeric( gsub("(\\d+)/(\\d+)", "\\1", af$V11, perl=T) ) / as.numeric ( gsub("(\\d+)/(\\d+)", "\\2", af$V11, perl=T) ) 
	

	summary(af)
	

	> > > > > > > > > > > >  V1               V2           V3               V4          V5
	 1:377853   Min.   :    2379   A:467482   Min.   :2   G/A    :349255
	 2:300277   1st Qu.:10387541   C:451939   1st Qu.:2   C/T    :347958
	 3:365262   Median :24633377   G:453036   Median :2   T/C    :343963
	 4:351117   Mean   :24004467   T:467070   Mean   :2   A/G    :343766
	 5:330512   3rd Qu.:37243744              3rd Qu.:2   A/T    : 77055
	 X:114506   Max.   :51822498              Max.   :2   T/A    : 76503
	                                                      (Other):301027
	       V6         V7          V8         V9              V10
	 Min.   :0   pop   :1578641   A:469511   A:485261   Min.   :0.4545
	 1st Qu.:0   rc    :     58   C:450250   C:435310   1st Qu.:0.7356
	 Median :0   rc|pop: 260828   G:451357   G:434498   Median :0.8800
	 Mean   :0                    T:468409   T:484458   Mean   :0.8313
	 3rd Qu.:0                                          3rd Qu.:0.9487
	 Max.   :0                                          Max.   :0.9850
	

	      V11
	 Min.   :0.01504
	 1st Qu.:0.05085
	 Median :0.11927
	 Mean   :0.16832
	 3rd Qu.:0.26389
	 Max.   :0.50000
	

	

	To put headers on the columns....
	

	af$"UGA:MajorAF"<-af$V10
	af$"UGA:MinorAF"<-af$V11
	

	write.table(af,"UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt",quote = FALSE,row.names = FALSE,sep = "\t")
	

	#To reduce the file size remove the unwanted columns!
	

	awk '{print $1 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 "\t" $12 "\t" $13}' UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt > B_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	#We want SNPs relative to the reference of >0.4 frequency in the population.
	#So, have it such that have pop changes, with the major allele the same as the reference allele and rc|pop changes with the minor 
	#allele the same as the reference allele.
	#So, to select just the rc|pop and then to take those with the major allele > 0.4. 
	

	find . -name "B_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt" | xargs grep -E 'rc' > ALT_RC_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	awk '$10 >0.4' ALT_RC_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt > 0.4_ALT_RC_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt
	Note, the above line may be fairly pointless to run as the major allele is going to be >/=0.5 anyway.... however.
	

	#Then to get when the minor allele is >0.4. 
	

	find . -name "B_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt" | xargs grep -v 'rc' > SAME_RA_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	awk '$11 >0.4' SAME_RA_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt > 0.4_SAME_RA_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	Great, so now have, for UGA MAPQ20, all the sites with a SNP allele different to the reference sequence with an allele frequency 
	of >0.4.
	

	cat 0.4_SAME_RA_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt 0.4_ALT_RC_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt > all_0.4_UGA_WBPS15_MAPQ20_AF
	

	Looks good - has the rc below the pop
	

	wc -l *0.4*
	Old transcriptome:
	  230350 0.4_ALT_RC_UGA_MAPQ20_biallelic_AF_freq.txt
	   88549 0.4_SAME_RA_UGA_MAPQ20_biallelic_AF_freq.txt
	  318899 all_0.4_UGA_MAPQ20_AF
	

	New transcriptome:
	  260886 0.4_ALT_RC_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt
	   99468 0.4_SAME_RA_UGA_WBPS15_MAPQ20_biallelic_AF_freq.txt
	  360354 all_0.4_UGA_WBPS15_MAPQ20_AF
	

	

	Need to make the file into a bed file to merge with the genes file. 
	

	awk '{print $1 "\t" ($2-1) "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 "\t" $10 "\t" $11}' all_0.4_UGA_WBPS15_MAPQ20_AF > all_0.4_UGA_WBPS15_MAPQ20_AF.bed
	

	Great - remember that the SNP is the THIRD column of the file!!! 
	

	To remove the header line....
	

	find . -name "all_0.4_UGA_WBPS15_MAPQ20_AF.bed" | xargs grep -v 'V' > B_all_0.4_UGA_WBPS15_MAPQ20_AF.bed
	mv B_all_0.4_UGA_WBPS15_MAPQ20_AF.bed all_0.4_UGA_WBPS15_MAPQ20_AF.bed
	

	

	Ok, so now have a file with all the SNPs of UGA cf the reference, with AF >0.4 and a file with all the CDS of all the genes. 
	To intersect these two files. 
	

	bedtools intersect -a all_0.4_UGA_WBPS15_MAPQ20_AF.bed -b gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.bed -wa -wb -f 2E-9 > CDS_all_0.4_UGA_WBPS15_MAPQ20_AF.bed
	

	Great. So to now try to count the number of SNPs per gene.... 
	

	awk -F "\t" '{a[$17] += $5; ++c[$17]} END{for (i in a) print i, a[i], c[i]}' CDS_all_0.4_UGA_WBPS15_MAPQ20_AF.bed > SNP_NUMBER_CDS_all_0.4_UGA_WBPS15_MAPQ20_AF
	

	Perfect. (the first col = gene name, the next = sum of the SNP count col..... ie. as biallelic this is double the actual number of SNPs, and the third col is the number of SNPs. 
	

	To make into a better file: 
	

	awk '{print $1 "\t" $3}' SNP_NUMBER_CDS_all_0.4_UGA_WBPS15_MAPQ20_AF > SNP_COUNT_CDS_all_0.4_UGA_WBPS15_MAPQ20_AF
	

	sort $1 SNP_COUNT_CDS_all_0.4_UGA_WBPS15_MAPQ20_AF > Sorted_SNP_COUNT_CDS_all_0.4_UGA_WBPS15_MAPQ20_AF
	

	sort $1 SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3 > Sorted_SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3
	

	Note, the SUM.gene file is delimited by white space. Need it to be tab delimited.... 
	

	awk -v OFS='\t' '{ $1=$1; print }' Sorted_SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3 > Sorted_SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3_Tab
	

	To combine the SNP total per CDS  and the length of the CDS:
	

	./list.join2List.pl Sorted_SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3_Tab Sorted_SNP_COUNT_CDS_all_0.4_UGA_WBPS15_MAPQ20_AF > UGA_WBPS15_MAPQ20_SNP_CDS_0.4
	

	So... to calculate the SNP RATE per CDS (SNP count/TOTAL sites)
	Col 1 = Gene name
	Col 2 = Sum of sites
	Col 3 = Sum of exons
	Col 4 = Gene name
	Col 5 = Sum of SNPs
	

	Therefore:
	

	awk '{print $1 "\t" $ 2 "\t" $3 "\t" $5 "\t" ($5/$2)}' UGA_WBPS15_MAPQ20_SNP_CDS_0.4 > SNP_RATE_UGA_WBPS15_MAPQ20_SNP_CDS_0.4
	

	

	################################################################################################################################
	######ISE - MAPQ20 ##########################################################################################################
	

	af<-read.table("biallelic_ISE_CDS_WBPS15_MAPQ20._rc", header=F, sep="\t", na.strings="")
	summary(af)
	

	  V1               V2           V3               V4          V5
	 1:269916   Min.   :    2379   A:320377   Min.   :2   G/A    :235022
	 2:197909   1st Qu.:10364182   C:308485   1st Qu.:2   C/T    :234624
	 3:263073   Median :24041981   G:308840   Median :2   T/C    :232833
	 4:242748   Mean   :23847734   T:320504   Mean   :2   A/G    :232190
	 5:214049   3rd Qu.:37141130              3rd Qu.:2   A/T    : 54161
	 X: 70511   Max.   :51822498              Max.   :2   T/A    : 53782
	                                                      (Other):215594
	       V6         V7          V8         V9              V10
	 Min.   :0   pop   :1039540   A:322471   A:330081   83/85  :   1619
	 1st Qu.:0   rc    :     21   C:306673   C:299781   85/87  :   1595
	 Median :0   rc|pop: 218645   G:307143   G:298266   92/94  :   1588
	 Mean   :0                    T:321919   T:330078   82/84  :   1582
	 3rd Qu.:0                                          79/81  :   1581
	 Max.   :0                                          87/89  :   1574
	                                                    (Other):1248667
	      V11
	 2/85   :   1653
	 2/87   :   1631
	 2/94   :   1625
	 2/81   :   1616
	 2/84   :   1610
	 2/89   :   1609
	 (Other):1248462
	

	

	

	To make the major and minor allele frequencies numerical and not factors...
	

	So, first:
	af$V10<-as.character(af$V10)
	af$V10<-as.numeric( gsub("(\\d+)/(\\d+)", "\\1", af$V10, perl=T) ) / as.numeric ( gsub("(\\d+)/(\\d+)", "\\2", af$V10, perl=T) ) 
	

	af$V11<-as.character(af$V11)
	af$V11<-as.numeric( gsub("(\\d+)/(\\d+)", "\\1", af$V11, perl=T) ) / as.numeric ( gsub("(\\d+)/(\\d+)", "\\2", af$V11, perl=T) ) 
	

	summary(af)
	af$"ISE:MajorAF"<-af$V10
	af$"ISE:MinorAF"<-af$V11
	

	 V1               V2           V3               V4          V5
	 1:269916   Min.   :    2379   A:320377   Min.   :2   G/A    :235022
	 2:197909   1st Qu.:10364182   C:308485   1st Qu.:2   C/T    :234624
	 3:263073   Median :24041981   G:308840   Median :2   T/C    :232833
	 4:242748   Mean   :23847734   T:320504   Mean   :2   A/G    :232190
	 5:214049   3rd Qu.:37141130              3rd Qu.:2   A/T    : 54161
	 X: 70511   Max.   :51822498              Max.   :2   T/A    : 53782
	                                                      (Other):215594
	       V6         V7          V8         V9              V10
	 Min.   :0   pop   :1039540   A:322471   A:330081   Min.   :0.4545
	 1st Qu.:0   rc    :     21   C:306673   C:299781   1st Qu.:0.6857
	 Median :0   rc|pop: 218645   G:307143   G:298266   Median :0.8293
	 Mean   :0                    T:321919   T:330078   Mean   :0.7978
	 3rd Qu.:0                                          3rd Qu.:0.9231
	 Max.   :0                                          Max.   :0.9846
	

	      V11           ISE:MajorAF      ISE:MinorAF
	 Min.   :0.01538   Min.   :0.4545   Min.   :0.01538
	 1st Qu.:0.07692   1st Qu.:0.6857   1st Qu.:0.07692
	 Median :0.17045   Median :0.8293   Median :0.17045
	 Mean   :0.20203   Mean   :0.7978   Mean   :0.20203
	 3rd Qu.:0.31395   3rd Qu.:0.9231   3rd Qu.:0.31395
	 Max.   :0.50000   Max.   :0.9846   Max.   :0.50000
	

	

	

	write.table(af,"ISE_WBPS15_biallelic_AF_freq.txt",quote = FALSE,row.names = FALSE,sep = "\t")
	

	

	#To reduce the file size remove the unwanted columns!
	

	awk '{print $1 "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 "\t" $12 "\t" $13}' ISE_WBPS15_biallelic_AF_freq.txt > B_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	#So, have it such that have pop changes, with the major allele the same as the reference allele and rc|pop changes with the minor allele the same as the reference allele.
	#So, to select just hte rc|pop and then to take those with the major allele > 0.4. 
	

	find . -name "B_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt" | xargs grep -E 'rc' > ALT_RC_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	awk '$10 >0.4' ALT_RC_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt > 0.4_ALT_RC_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	#Then to get when the minor allele is >0.4. 
	

	find . -name "B_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt" | xargs grep -v 'rc' > SAME_RA_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	awk '$11 >0.4' SAME_RA_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt > 0.4_SAME_RA_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt
	

	Great, so now have, for ISE MAPQ20, all the sites with a SNP allele different to the reference sequence with an allele frequency of >0.4.
	

	cat 0.4_SAME_RA_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt 0.4_ALT_RC_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt > all_0.4_ISE_WBPS15_MAPQ20_AF
	

	Looks good - has the rc below the pop
	

	wc -l *0.4*ISE*
	Old transcriptome:
	  192396 0.4_ALT_RC_ISE_MAPQ20_biallelic_AF_freq.txt
	   80929 0.4_SAME_RA_ISE_MAPQ20_biallelic_AF_freq.txt
	  273325 all_0.4_ISE_MAPQ20_AF
	

	New transcriptome:
	  218666 0.4_ALT_RC_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt
	   91595 0.4_SAME_RA_ISE_WBPS15_MAPQ20_biallelic_AF_freq.txt
	  310261 all_0.4_ISE_WBPS15_MAPQ20_AF
	

	Need to make the file into a bed file to merge with the genes file. 
	

	awk '{print $1 "\t" ($2-1) "\t" $2 "\t" $3 "\t" $4 "\t" $5 "\t" $6 "\t" $7 "\t" $8 "\t" $9 "\t" $10 "\t" $11}' all_0.4_ISE_WBPS15_MAPQ20_AF > all_0.4_ISE_WBPS15_MAPQ20_AF.bed
	

	Great - remember that the SNP is the THIRD column of the file!!! 
	

	To remove the header line....
	

	find . -name "all_0.4_ISE_WBPS15_MAPQ20_AF.bed" | xargs grep -v 'V' > B_all_0.4_ISE_WBPS15_MAPQ20_AF.bed
	mv B_all_0.4_ISE_WBPS15_MAPQ20_AF.bed all_0.4_ISE_WBPS15_MAPQ20_AF.bed
	

	

	Ok, so now have a file with all the SNPs of ISE cf the reference, with AF >0.4 and a file with all the CDS of all the genes. 
	To intersect these two files. 
	

	bedtools intersect -a all_0.4_ISE_WBPS15_MAPQ20_AF.bed -b gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.bed -wa -wb -f 2E-9 > CDS_all_0.4_ISE_WBPS15_MAPQ20_AF.bed
	

	Great. So to now try to count the number of SNPs per gene.... 
	

	awk -F "\t" '{a[$17] += $5; ++c[$17]} END{for (i in a) print i, a[i], c[i]}' CDS_all_0.4_ISE_WBPS15_MAPQ20_AF.bed > SNP_NUMBER_CDS_all_0.4_ISE_WBPS15_MAPQ20_AF
	

	Perfect. (the first col = gene name, the next = sum of the SNP count col..... ie. as biallelic this is double the actual number of SNPs, and the third col is the number of SNPs. 
	

	To make into a better file: 
	

	awk '{print $1 "\t" $3}' SNP_NUMBER_CDS_all_0.4_ISE_WBPS15_MAPQ20_AF > SNP_COUNT_CDS_all_0.4_ISE_WBPS15_MAPQ20_AF
	

	sort $1 SNP_COUNT_CDS_all_0.4_ISE_WBPS15_MAPQ20_AF > Sorted_SNP_COUNT_CDS_all_0.4_ISE_WBPS15_MAPQ20_AF
	

	

	./list.join2List.pl Sorted_SUM.gene.length.CDS.T1.haemonchus_contortus.PRJEB506.WBPS15.annotations.gff3_Tab Sorted_SNP_COUNT_CDS_all_0.4_ISE_WBPS15_MAPQ20_AF > ISE_WBPS15_MAPQ20_SNP_CDS_0.4
	

	So... to calculate the SNP RATE per CDS (SNP count/TOTAL sites)
	Col 1 = Gene name
	Col 2 = Sum of sites
	Col 3 = Sum of exons
	Col 4 = Gene name
	Col 5 = Sum of SNPs
	

	Therefore:
	

	awk '{print $1 "\t" $ 2 "\t" $3 "\t" $5 "\t" ($5/$2)}' ISE_WBPS15_MAPQ20_SNP_CDS_0.4 > SNP_RATE_ISE_WBPS15_MAPQ20_SNP_CDS_0.4
	

	#####PERFECT!! ######
	

	####To merge the UGA and ISE SNP_RATE files ######
	

	./list.join2List.pl SNP_RATE_ISE_WBPS15_MAPQ20_SNP_CDS_0.4 SNP_RATE_UGA_WBPS15_MAPQ20_SNP_CDS_0.4 > SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20
	

	Worked fine:
	

	awk '{print $1 "\t" $ 2 "\t" $3 "\t" $4 "\t" $5 "\t" "\t" $9 "\t" $10}' SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20 > B_SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20
	mv B_SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20 SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20
	

	Col 1 = Gene name
	Col 2 = Sum of sites
	Col 3 = Sum of exons
	Col 4 = ISE SNP count
	Col 5 = ISE CDS SNP rate (SNP count/Sum of sites)
	Col 6 = UGA SNP count
	Col 7 = UGA CDS SNP rate (SNP count/Sum of sites)
	

	

	#######################################################################################################################################
	######To merge with DE-Seq data #####################################################################################################
	

	Note, to combine with Roz's DESEQ data need to remove the -00001 from gene names!! 
	

	sed 's/-00001//g' SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20 > B_SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20
	

	All good. Will re-name the file: 
	mv B_SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20 SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20
	

	

	Then to merge with Roz DESEQ data:
	

	First have to remove the header line pf Roz's file, also removed the end line (in excel - had an incomplete line of totals):
	

	sort $1 summary_DESEQ_WBPS15gff_alpha0.01_males_all3IVMcomps.txt > Sorted.summary_DESEQ_WBPS15gff_alpha0.01_males_all3IVMcomps.txt
	

	./list.join2List.pl SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20 Sorted.summary_DESEQ_WBPS15gff_alpha0.01_males_all3IVMcomps.txt > summary_DESEQ_WBPS15gff_alpha0.01_males_all3IVMcomps_CDS_SNP_RATE_ISE_then_UGA_WBPS15_MAPQ20.txt
	

	

	In R:
	R version 3.3.3 (2017-03-06) -- "Another Canoe"
	

	t<-read.table("summary_DESEQ_WBPS15gff_alpha0.01_males_all3IVMcomps_CDS_SNP_RATE_ISE_then_UGA_WBPS15_MAPQ20.txt", header=F)
	summary(t)
	             V1              V2              V3                V4
	 HCON_00000020:    1   Min.   :    2   Min.   :  1.000   Min.   :   1.00
	 HCON_00000030:    1   1st Qu.:  494   1st Qu.:  3.000   1st Qu.:   4.00
	 HCON_00000040:    1   Median :  926   Median :  7.000   Median :  13.00
	 HCON_00000050:    1   Mean   : 1221   Mean   :  8.892   Mean   :  24.74
	 HCON_00000060:    1   3rd Qu.: 1516   3rd Qu.: 12.000   3rd Qu.:  30.00
	 HCON_00000070:    1   Max.   :22837   Max.   :138.000   Max.   :1099.00
	 (Other)      :21233                                     NA's   :6258
	       V5                 V6                V7                       V8
	 Min.   :0.000000   Min.   :   1.00   Min.   :0.000000   HCON_00000020:    1
	 1st Qu.:0.000000   1st Qu.:   5.00   1st Qu.:0.001912   HCON_00000030:    1
	 Median :0.006696   Median :  14.00   Median :0.011499   HCON_00000040:    1
	 Mean   :0.013494   Mean   :  25.26   Mean   :0.015707   HCON_00000050:    1
	 3rd Qu.:0.021497   3rd Qu.:  31.00   3rd Qu.:0.023850   HCON_00000060:    1
	 Max.   :0.207792   Max.   :1018.00   Max.   :0.189702   (Other)      :19656
	                    NA's   :4161                         NA's         : 1578
	       V9                 V10                V11              V12
	 Min.   :      0.0   Min.   :-10.0802   Min.   :0.0466   Min.   :-28.5693
	 1st Qu.:      5.8   1st Qu.: -0.4978   1st Qu.:0.1258   1st Qu.: -1.4016
	 Median :    176.6   Median :  0.0000   Median :0.2311   Median :  0.0000
	 Mean   :   2173.6   Mean   : -0.0859   Mean   :0.9011   Mean   : -0.0635
	 3rd Qu.:   1053.1   3rd Qu.:  0.3301   3rd Qu.:0.9332   3rd Qu.:  1.3318
	 Max.   :1352821.3   Max.   : 11.8382   Max.   :4.8568   Max.   : 24.6581
	 NA's   :1578        NA's   :2497       NA's   :2497     NA's   :2497
	      V13              V14                   V15             V16
	 Min.   :0.0000   Min.   :0.000   HCON_00000020:    1   Min.   :      0.0
	 1st Qu.:0.0050   1st Qu.:0.005   HCON_00000030:    1   1st Qu.:      5.8
	 Median :0.1701   Median :0.166   HCON_00000040:    1   Median :    176.6
	 Mean   :0.3175   Mean   :0.304   HCON_00000050:    1   Mean   :   2173.6
	 3rd Qu.:0.6000   3rd Qu.:0.579   HCON_00000060:    1   3rd Qu.:   1053.1
	 Max.   :1.0000   Max.   :1.000   (Other)      :19656   Max.   :1352821.3
	 NA's   :2527     NA's   :5801    NA's         : 1578   NA's   :1578
	      V17               V18              V19                V20
	 Min.   :-8.9181   Min.   :0.0466   Min.   :-21.7371   Min.   :0.0000
	 1st Qu.:-0.2456   1st Qu.:0.1245   1st Qu.: -0.8843   1st Qu.:0.0397
	 Median : 0.0222   Median :0.2265   Median :  0.0941   Median :0.3112
	 Mean   : 0.1469   Mean   :0.8714   Mean   :  0.1389   Mean   :0.3786
	 3rd Qu.: 0.3852   3rd Qu.:0.8584   3rd Qu.:  1.1086   3rd Qu.:0.6766
	 Max.   : 7.9698   Max.   :4.8568   Max.   : 13.2452   Max.   :1.0000
	 NA's   :2497      NA's   :2497     NA's   :2497       NA's   :2527
	      V21                   V22             V23                 V24
	 Min.   :0.000   HCON_00000020:    1   Min.   :      0.0   Min.   :-6.1910
	 1st Qu.:0.066   HCON_00000030:    1   1st Qu.:      5.8   1st Qu.:-0.2639
	 Median :0.405   HCON_00000040:    1   Median :    176.6   Median : 0.0147
	 Mean   :0.426   HCON_00000050:    1   Mean   :   2173.6   Mean   : 0.1769
	 3rd Qu.:0.748   HCON_00000060:    1   3rd Qu.:   1053.1   3rd Qu.: 0.3901
	 Max.   :1.000   (Other)      :19656   Max.   :1352821.3   Max.   : 8.4640
	 NA's   :6520    NA's         : 1578   NA's   :1578        NA's   :2497
	      V25              V26                V27              V28
	 Min.   :0.0466   Min.   :-26.8272   Min.   :0.0000   Min.   :0.000
	 1st Qu.:0.1251   1st Qu.: -1.0184   1st Qu.:0.0296   1st Qu.:0.056
	 Median :0.2275   Median :  0.0680   Median :0.2720   Median :0.355
	 Mean   :0.8802   Mean   :  0.0116   Mean   :0.3596   Mean   :0.403
	 3rd Qu.:0.8818   3rd Qu.:  1.1565   3rd Qu.:0.6492   3rd Qu.:0.722
	 Max.   :4.8568   Max.   : 18.3949   Max.   :1.0000   Max.   :1.000
	 NA's   :2497     NA's   :2497       NA's   :2527     NA's   :5801
	

	

	#######################################################################################################################################
	######To merge with FEMALE DE-Seq data #####################################################################################################
	

	23.09.20
	

	To merge with the female DESEQ data. 
	Excel is currently not working. So to try loading as sent and see what happens.
	

	

	First have to remove the header line pf Roz's file - did this in nano.
	

	For some reason all of Roz's gene names have "gene_name". So need to remove this!
	

	sed 's/"//g' summary_DESEQ_WBSP15_alpha0.01_females.txt > Changed.summary_DESEQ_WBSP15_alpha0.01_females.txt
	

	sort $1 Changed.summary_DESEQ_WBSP15_alpha0.01_females.txt > Sorted.summary_DESEQ_WBSP15_alpha0.01_females.txt
	

	./list.join2List.pl SNP_RATE_ISE_AND_UGA_SNP_CDS_0.4_WBPS15_MAPQ20 Sorted.summary_DESEQ_WBSP15_alpha0.01_females.txt > summary_DESEQ_WBSP15_alpha0.01_females_CDS_SNP_RATE_ISE_then_UGA_WBPS15_MAPQ20.txt

